python/ray/data/dataset.py

```diff
@@ -2320,7 +2321,7 @@ def iter_rows(self, *, prefetch_blocks: int = 0) -> Iterator[Union[T, TableRow]]
                 else "native"
             )
         for batch in self.iter_batches(
-            prefetch_blocks=prefetch_blocks, batch_format=batch_format
+            batch_size=None, prefetch_blocks=prefetch_blocks, batch_format=batch_format
```

Comment:
hmm why not use the default 4096 here?

I am kind of worried about different behavior between `iter_rows` (read entire block as one batch) vs `iter_batches` (read 4096 bytes as one batch) here. Given the interface of `iter_rows` just outputs row, so it should not matter to users when it comes to each batch size right?

If user is worried about performance, they should already use `iter_batches()` instead of `iter_rows()`.

Reply:
@c21 `.iter_rows()` provides a zero-copy row view of the underlying block, so such batching would provide an unnecessary extra slice step that currently results in a full copy of the batch, so it's much more efficient to grab the unsliced block and provide the zero-copy row view on that block.

It should also be noted that this `batch_size=4096` is a consumer-side slice, we still fetch the entire block regardless of the batch size, so specifying a batch size here will not improve memory stability and will only hurt performance without any benefit.

Reply:
> It should also be noted that this batch_size=4096 is a consumer-side slice, we still fetch the entire block regardless of the batch size

@clarkzinzow - I see. That makes sense to me now. Thanks for explanation.